pkg/deprecatetemplates/enforcer_test.go

```diff
@@ -0,0 +1,158 @@
+package deprecatetemplates
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+func TestLoadTemplates(t *testing.T) {
+	testcases := []struct {
+		description string
+		updaterCfg  plugins.ConfigUpdater
+		expected    sets.String
+	}{
+		{
+			description: "template is detected",
+			updaterCfg: plugins.ConfigUpdater{
+				Maps: map[string]plugins.ConfigMapSpec{
+					"ci-operator/templates/this-is-a-template.yaml": {Name: "template"},
+				},
+			},
+			expected: sets.NewString("template"),
+		},
+		{
+			description: "not a template is ignored",
+			updaterCfg: plugins.ConfigUpdater{
+				Maps: map[string]plugins.ConfigMapSpec{
+					"ci-operator/config/this/is-not/a-template.yaml": {Name: "not-a-template"},
+				},
+			},
+			expected: sets.NewString(),
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.description, func(t *testing.T) {
+			enforcer := Enforcer{}
+			enforcer.LoadTemplates(tc.updaterCfg)
+			if diff := cmp.Diff(tc.expected, enforcer.existingTemplates); diff != "" {
+				t.Errorf("%s: templates differ from expected:\n%s", tc.description, diff)
+			}
+		})
+	}
+}
+
+type mockAllowlist struct {
+	jobs map[string]sets.String
+}
+
+func (m *mockAllowlist) Insert(job config.JobBase, template string) {
+	if _, ok := m.jobs[template]; !ok {
+		m.jobs[template] = sets.NewString()
+	}
+	m.jobs[template].Insert(job.Name)
+}
+
+func (m *mockAllowlist) Save(_ string) error {
+	panic("this should never be called")
+}
+
+type mockJobConfig struct {
+	presubmits  []config.Presubmit
+	postsubmits []config.Postsubmit
+	periodics   []config.Periodic
+}
+
+func (m *mockJobConfig) AllStaticPostsubmits(_ []string) []config.Postsubmit {
+	return append([]config.Postsubmit{}, m.postsubmits...)
+}
+func (m *mockJobConfig) AllStaticPresubmits(_ []string) []config.Presubmit {
+	return append([]config.Presubmit{}, m.presubmits...)
```

Comment:
Why the append? To get a new slice?

Reply:
Yeah, I wanted the mock to vend out copies, to prevent any possible interference.